libv4l not compiling with libstdcpp

Maintainer: @thess 
Environment: mvebu, libstdcpp

Log:
```
make[8]: Entering directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3/utils/media-ctl'
  CC       libmediactl_la-libmediactl.lo
  CCLD     libmediactl.la
/home/mangix/devstuff/openwrt/staging_dir/toolchain-arm_cortex-a9+vfpv3_gcc-7.4.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/7.4.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CC       libv4l2subdev_la-libv4l2subdev.lo
  CCLD     libv4l2subdev.la
/home/mangix/devstuff/openwrt/staging_dir/toolchain-arm_cortex-a9+vfpv3_gcc-7.4.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/7.4.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CC       media-ctl.o
  CC       options.o
  CCLD     media-ctl
make[8]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3/utils/media-ctl'
make[7]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3/utils/media-ctl'
Making all in v4l2-compliance
make[7]: Entering directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3/utils/v4l2-compliance'
  CXX      v4l2_compliance-v4l2-compliance.o
  CXX      v4l2_compliance-v4l2-test-debug.o
  CXX      v4l2_compliance-v4l2-test-input-output.o
  CXX      v4l2_compliance-v4l2-test-controls.o
  CXX      v4l2_compliance-v4l2-test-io-config.o
  CXX      v4l2_compliance-v4l2-test-formats.o
  CXX      v4l2_compliance-v4l2-test-buffers.o
  CXX      v4l2_compliance-v4l2-test-codecs.o
  CXX      v4l2_compliance-v4l2-test-subdevs.o
  CXX      v4l2_compliance-v4l2-test-media.o
  CXX      v4l2_compliance-v4l2-test-colors.o
  CXX      v4l2_compliance-media-info.o
media-info.cpp: In function 'media_type mi_media_detect_type(const char*)':
media-info.cpp:68:28: error: variable 'std::ifstream uevent_file' has initializer but incomplete type
  std::ifstream uevent_file(uevent_path.c_str());
                            ^~~~~~~~~~~
media-info.cpp: In function 'std::__cxx11::string mi_media_get_device(__u32, __u32)':
media-info.cpp:109:28: error: variable 'std::ifstream uevent_file' has initializer but incomplete type
  std::ifstream uevent_file(uevent_path.c_str());
                            ^~~~~~~~~~~
Makefile:762: recipe for target 'v4l2_compliance-media-info.o' failed
make[7]: *** [v4l2_compliance-media-info.o] Error 1
make[7]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3/utils/v4l2-compliance'
Makefile:467: recipe for target 'all-recursive' failed
make[6]: *** [all-recursive] Error 1
make[6]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3/utils'
Makefile:580: recipe for target 'all-recursive' failed
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/v4l-utils-1.16.3'
Makefile:507: recipe for target 'all' failed
make[4]: *** [all] Error 2
```

--- a/libs/libv4l/Makefile
+++ b/libs/libv4l/Makefile
@@ -6,12 +6,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=v4l-utils
-PKG_VERSION:=1.16.5
+PKG_VERSION:=1.16.6
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=https://www.linuxtv.org/downloads/v4l-utils
-PKG_HASH:=ed80242510385017a1dc566e17a285a77222bb301f5bc19386badfcc2c19df1b
+PKG_HASH:=f9dac1878e3d5636eab7f56bb209fdfc66b94ee8a2aae54dcb4282fe63a678ae
 
 PKG_MAINTAINER:=Ted Hess <thess@kitschensync.net>
 PKG_LICENSE:=GPL-2.0 LGPL-2.1

--- a/libs/libv4l/patches/020-add-missing-includes.patch
+++ b/libs/libv4l/patches/020-add-missing-includes.patch
@@ -20,11 +20,10 @@
  #include <linux/cec-funcs.h>
 --- a/utils/common/media-info.cpp
 +++ b/utils/common/media-info.cpp
-@@ -20,7 +20,7 @@
- 
+@@ -21,6 +21,7 @@
  #include <linux/media.h>
  
--#include <fstream>
+ #include <fstream>
 +#include <iostream>
  #include <media-info.h>
  
@@ -42,22 +41,20 @@
  #include <map>
 --- a/utils/v4l2-compliance/media-info.cpp
 +++ b/utils/v4l2-compliance/media-info.cpp
-@@ -20,7 +20,7 @@
- 
+@@ -21,6 +21,7 @@
  #include <linux/media.h>
  
--#include <fstream>
+ #include <fstream>
 +#include <iostream>
  #include <media-info.h>
  
  static std::string num2s(unsigned num, bool is_hex = true)
 --- a/utils/v4l2-ctl/media-info.cpp
 +++ b/utils/v4l2-ctl/media-info.cpp
-@@ -20,7 +20,7 @@
- 
+@@ -21,6 +21,7 @@
  #include <linux/media.h>
  
--#include <fstream>
+ #include <fstream>
 +#include <iostream>
  #include <media-info.h>
  

--- a/libs/libv4l/patches/030-getsubopt.patch
+++ b/libs/libv4l/patches/030-getsubopt.patch
@@ -3,11 +3,9 @@ musl libs will set value to NULL which leads to crash.
 
 Simply avoid getsubopt, since we cannot rely on it.
 
-diff --git a/utils/v4l2-ctl/v4l2-ctl-common.cpp b/utils/v4l2-ctl/v4l2-ctl-common.cpp
-index 3ea6cd3..291fb3e 100644
 --- a/utils/v4l2-ctl/v4l2-ctl-common.cpp
 +++ b/utils/v4l2-ctl/v4l2-ctl-common.cpp
-@@ -692,16 +692,17 @@ static bool parse_subset(char *optarg)
+@@ -679,16 +679,17 @@ static bool parse_subset(char *optarg)
  
  static bool parse_next_subopt(char **subs, char **value)
  {

